_overviews/overview/scala-features.md

```diff
@@ -0,0 +1,496 @@
+  - Classes and abstract types as object members
+  - Structural types
+  - Path-dependent types
+  - Compound types
+  - Explicitly typed self references
+  - Generic classes
+  - Polymorphic methods
+  - Upper and lower type bounds
+  - Variance (covariance and contravariance)
+  - Annotation
+  - Views
+  - Type inference
+  - “Type enrichment” (implicits)
+  - Statically typed
+  - ADTs
+
+- Possible resources
+    - https://en.wikipedia.org/wiki/Scala_(programming_language)#Expressive_type_system
+    - https://docs.scala-lang.org/tour/tour-of-scala.html
+    - https://www.artima.com/scalazine/articles/scalas_type_system.html
+    - https://heather.miller.am/blog/types-in-scala.html
+
+- DOT, Soundness
+    - http://dotty.epfl.ch/blog/2016/02/17/scaling-dot-soundness
+    - https://www.scala-lang.org/blog/2016/02/03/essence-of-scala.html
+-->
+
+
+### It’s a pure functional programming language
+
+Scala is a functional programming (FP) language, meaning:
+
+- Functions are variables, and can be passed around like any other variable
+- Higher-order functions are directly supported
+- Lambdas are built in
+- Everything in Scala is an expression that returns a value
+- Syntactically it’s easy to use immutable variables, and their use is encouraged
+- It has a wealth of immutable collections classes in the standard library
+- Those collections classes come with dozens of functional methods: they don’t mutate the collection, but instead return an updated copy of the data
+
+
+### It’s a pure object-oriented programming language
+
+Scala is a _pure_ object-oriented programming (OOP) language. Every variable is an object, and every “operator” is a method.
+
+In Scala, all types inherit from a top-level class `Any`, whose immediate children are `AnyVal` (_value types_, such as `Int` and `Boolean`) and `AnyRef` (_reference types_, as in Java). This means that the Java distinction between primitive types and boxed types (e.g. `int` vs. `Integer`) isn’t present in Scala. Boxing and unboxing is completely transparent to the user.
+
+<!-- TODO: add the “hierarchy” image here -->
+
+
+### Scala supports the fusion of FP and OOP
+
+<!-- from a slide here: https://twitter.com/alexelcu/status/996408359514525696 -->
+
+The essence of Scala is the fusion of functional programming and object-oriented programming in a typed settings:
+
+- Functions for the logic
+- Objects for the modularity
+
+As [Martin Odersky has stated](https://jaxenter.com/current-state-scala-odersky-interview-129495.html), “Scala was designed to show that a fusion of functional and object-oriented programming is possible and practical.”
+
+For more details on this philosophy, see the TODO section in the Reference documentation.
+
+
+### Contextual abstractions provide a clear way to implement term inference
+
+Following Haskell, Scala was the second popular language to have some form of _implicits_. In Scala 3 these concepts have been completely re-thought and more clearly implemented.
```

Comment:
Don't say "more clearly" this is offensive to Haskell developers. Maybe "integrated into the language in a conceptually sound way".